Source/WebKit/Scripts/generate-plain-cpp-swift-interop-header.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+set -e
+
+# Swift/C++ interop generates a header file to be included in C++ code that provides
+# definitions of types encountered in Swift.
+# There's built-in Xcode functionality to generate this file.
+# But we need to do it twice:
+# - once, processing Swift code that pertains to objective-C[++] types to generate
+#   a header for inclusion in objective-C++ code
+# - once, processing only Swift code that does not involve objective-C[++] types,
+#   to generate a header which can be safely included in pure C++ code.
+# This shell script does the second thing. We use the built-in XCode action
+# for the first.
+# Working around:
+# - rdar://152836730
+# - rdar://158843666
+# We will be able to remove this script entirely once these issues are fixed.
+# Known deficiencies in thie script:
+# - We do not attempt to provide all the #defines normally provided to C++ nor
+#   the -Ds to Swift.
+# - We do not provide a full include path.
+# - There is no attempt at full dependency tracking (.d files etc.)
+# This is mostly OK because right now the only header file included from Swift
+# refers to only one other header.
+
+xcrun swiftc -typecheck -emit-clang-header-path "${SCRIPT_OUTPUT_FILE_0}" "${SCRIPT_INPUT_FILE_1}" -DENABLE_SWIFT_DEMO_URI_SCHEME \
```

Comment:
Could we pass in all of OTHER_SWIFT_FLAGS here?
```suggestion
xcrun swiftc -typecheck -emit-clang-header-path "${SCRIPT_OUTPUT_FILE_0}" "${SCRIPT_INPUT_FILE_1}" ${OTHER_SWIFT_FLAGS} \
```

Among other things, that will give access to the full set of Platform.h defines.

Reply:
This didn't work for some reason - it didn't contain `-DENABLE_SWIFT_DEMO_URI_SCHEME` but it did seem to contain other flags which confused the swiftc command-line parsing.

How much do you care about this? IMO we _want_ this script to be primitive and limited so that people aren't tempted to use it, before we replace it with the built-in Xcode functionality when the known bugs are fixed. I don't want people using this beyond the Swift demo logo use-case, so this limitation seems to me to be actively a good thing.

Reply:
Strange, I wonder if the `generate-platform-args` script is not parsing GCC_PREPROCESSOR_DEFINITIONS like we'd expect.

But yeah, I agree that this isn't so important that it needs to be addressed. We can treat this script as a hacky workaround for ENABLE_SWIFT_DEMO_URI_SCHEME, and we can extend it further if changing Swift doesn't pan out and we're stuck with this approach.

Reply:
I looked into it. `WebKitBuild/Debug/DerivedSources/WebKit/platform-enabled-swift-args.arm64e.resp` does contain `-DENABLE_SWIFT_DEMO_URI_SCHEME`. However `OTHER_SWIFT_FLAGS` actually asks this script to `@` include `WebKitBuild/Debug/DerivedSources/WebKit/platform-enabled-swift-args.undefined_arch.resp` which doesn't exist at all.

I thought that adding `runOncePerArchitecture = 1` to the relevant part of the `pbxproj` might solve this, but it doesn't seem to. In any case the built-in Xcode functionality to emit such a header does not seem to be architecture-specific so I'm not keen on making an arch-specific header file.

Also, the specific error I see is:

```
error: unexpected input file: @/path/WebKitBuild/Debug/DerivedSources/WebKit/platform-enabled-swift-args.undefined_arch.resp
```

which suggests that `swiftc` in this mode doesn't even support `resp` files.

If you know a magic way to fix all this let me know and I'm happy to do in a follow-up PR. Marking as resolved meanwhile.